packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -1722,7 +1742,10 @@ def reset_tendermint_with_wait(
                     for handler_name in self.context.handlers.__dict__.keys():
                         dialogues = getattr(self.context, f"{handler_name}_dialogues")
                         dialogues.cleanup()
-
+                    # in case of successful reset we store the reset params in the shared state,
+                    # so that in the future if the communication with tendermint breaks, and we need to
+                    # perform a hard reset to restore it, we can use these as the right ones
+                    self.context.shared_state["reset_params"] = reset_params
```

Comment:
`shared_state` is not conceptually the right place. There we store information which is relevant cross skills. I'd save it in a model on the skill.

Reply:
@angrybayblade could rename `shared_state` for v2 and to call it `skills_stared_state` or similar. It gets often misinterpreted.

Reply:
https://github.com/valory-xyz/open-aea/issues/470